src/java/htsjdk/samtools/BAMQueryMultipleIntervalsIteratorFilter.java

```diff
@@ -0,0 +1,59 @@
+package htsjdk.samtools;
+
+import htsjdk.samtools.util.CoordMath;
+
+/**
+ * Filters out records that do not match any of the given intervals and query type.
+ */
+public class BAMQueryMultipleIntervalsIteratorFilter implements BAMIteratorFilter {
```

Comment:
Since you've made this formerly private class public, can you add direct unit tests for it?